### PR TITLE
Make activation function addition in conv2d and upconv2d dynamic

### PR DESCRIPTION
**Pull request by poad42**
at lines 73,74,79,80 and 92 and probably others set activation to be linear but in reality Relu gets added as there is no check for linear.

### CODE REVIEW

**Comment by zcemycl** [comment]
Sorry, this does not work. One is for professional programming, no one uses eval much. Another is that, if that code gets executed, it will fail to find that function, since LeakyRelu is not leakyrelu, ReLu is not relu.

**Comment by poad42** [comment]
Yup I made a mistake there, but the point of the original PR still stands, I looked a bit more on this and tf.keras.layers.linear, and doesn't actually exist

so having a call like conv2d(dim=1, size=1, act="linear") or upconv2d(dim=d, act="linear") doesn't make any sense.

**Comment by zcemycl** [comment]
act="linear" will just pass the if statement, since no activation function is applied.

**Comment by poad42** [comment]
I realize it does that, but isn't that just bad practice? Why not have default to be None, and pass Relu explicitly if neeeded?

**Comment by zcemycl** [comment]
Python can allow you to pass function handle in it, but not everyone does that. I don't want to overcomplicate the program. 

Also, when you pass function handle multiple times, it can confuse the back propagation of neural network, since it may overcomplicate the computation graph, and backprop may not know how to backprop.  

You can of course use else statement and assert 'Not implemeted error'. But I don't support it, programming isn't perfect, and relu and leakyrelu are common enough.
